nixos.nix

```diff
@@ -57,29 +72,44 @@ in
   };
 
   config = {
-    environment.etc =
+    systemd.services =
       let
-        link = file:
-          pkgs.runCommand
-            "${replaceStrings [ "/" "." " " ] [ "-" "" "" ] file}"
-            { }
-            "ln -s '${file}' $out";
-
-        # Create environment.etc link entry.
-        mkLinkNameValuePair = persistentStoragePath: file: {
-          name = removePrefix "/etc/" file;
-          value = { source = link (concatPaths [ persistentStoragePath file ]); };
-        };
+        mkBindOrLinkFileService = persistentStoragePath: file:
```

Comment:
Nit, but I think all references to `BindOrLink` should be replaced with `Persist` to align with the new prefix.